proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +111,335 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
-the input vector. For the packed case, the number of components in the input vector must large enough to hold the *K*
-packed components.
 
-*Layout* is an enum value, `DXILMatrixLayout`. Optimal layouts are opaque implementation specific layouts, the D3D call
- `CooperativeVectorConvertMatrix` can be used to convert the *Matrix* to an optimal layout. Row-Major and Column-Major
- layouts are also supported.
 
-The *Transposed* parameter indicates if the *Matrix* is transposed before performing the multiply. In linear algebra,
-the[transpose](https://en.wikipedia.org/wiki/Transpose) of a matrix is an operator which flips a matrix over its
-diagonal; that is, it switches the row and column indices of the matrix. Transposing is not supported for the
-RowMajor/ColumnMajor layouts. Not all component types support transposing. It is left to implementations to define
-which types support matrix transposing. "TransposeSupported" flag from the [CheckFeatureSupport]
-(#check-feature-support) struct is used to determine if a matrix transpose is supported. Note that even for the
-type/interpretation combinations with guaranteed [support](#minimum-support-set), transpose support isn't guaranteed
-and needs to be checked explicitly.
+> TODO: consider using a different set of interpretation values for in-memory
+> interpretations versus vector interpretations.
 
-**Type Interpretations**
 
-The types of *InputVector*, *Matrix* and *BiasVector* are all determined by their respective interpretation parameters.
-For the Matrix and BiasVector which are stored in (RW)ByteAddressBuffers, this is straightforward: the *M*
-and *K* *VecMatOpDescriptor* parameters describe the dimensions of the *Matrix*/*BiasVector*, these are loaded from the
-offsets *MatrixOffset* and *Biasoffset* respectively and the *Mi* and *Bi* parameters which
-are *DXILTypeInterpretation* enums specify the element type.
 
-*InputVector* is an HLSL vectors of a given type *InputTy* . However, the type that the elements of this vector are
- interpreted as in the matrix-vector operation is specified by the *InputInterpretation* parameter. The reason is that
- the interpretation parameter allows the elements to be interpreted as types not natively supported by HLSL, e.g.
- uint8/sint8. 
+##### Bias Matrix
 
-The legal conversions from the declared *InputType* to *InputInterpretation: Ii* and the
-corresponding *MatrixInterpretation: Mi* and *BiasInterpretation: Bi* are implementation dependent and can be queried.
-See[CheckFeatureSupport](#check-feature-support) section for details. An exception to this rule is the set of
-combinations guaranteed to be supported on all devices supporting this feature. See [Minimum Support Set]
-(#minimum-support-set).  Note that *Transposed* is always queried.
+> * TODO: alignment requirements for **bias vector offset**
+> * TODO: are packed types allowed for bias vectors?
```

Comment:
I think that this actual discusses alignment requirements now.

Packed types aren't allowed, right?

Reply:
@anupamachandra - can you confirm that packed types aren't allowed for bias vectors loaded from memory please?

Reply:
Oh, I see you have!